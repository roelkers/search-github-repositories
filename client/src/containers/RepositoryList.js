import React from 'react';
import { connect } from 'react-redux';
import Repository from '../components/Repository';
import { addBookmark } from "../actions";

export function RepositoryList({ repositories, onAddBookmark }) {
  const repositoriesJsx = repositories.map(repository => {
    return (
      <Repository repository={repository} onAddBookmark={onAddBookmark} key={repository.id} />
    );
  })
  return ( 
    <div>
      {repositoriesJsx.length>0? <h2 className='listHeading'>Repositories: </h2> : ''}
      {repositoriesJsx}
    </div>
  );
}

const mapStateToProps = state => {
  return {
    repositories: state
  };
};

const mapDispatchToProps = dispatch => {
  return {
    onAddBookmark: id => {
      dispatch(addBookmark(id));
    },
  };
};



export default connect(
  mapStateToProps,
  mapDispatchToProps
)(RepositoryList);