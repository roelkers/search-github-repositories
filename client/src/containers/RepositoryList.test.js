import React from 'react';
import {RepositoryList} from './RepositoryList';
import renderer from 'react-test-renderer';

describe('Repository', () => {
    const repositories = [{
        id : 1,
        name : "test",
        owner : "test_owner",
        stars : 2,
        forks : 3
    }];
    it('should match the snapshot', () => {
        const component = renderer.create(<RepositoryList repositories={repositories} />);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })
});