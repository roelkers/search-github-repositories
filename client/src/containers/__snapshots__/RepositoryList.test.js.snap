// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Repository should match the snapshot 1`] = `
<div>
  <h2
    className="listHeading"
  >
    Repositories: 
  </h2>
  <div
    className="repository"
  >
    <h2>
      test
    </h2>
    <p>
      Id: 
      1
    </p>
    <p>
      Owner: 
      test_owner
    </p>
    <p>
      Forks: 
      3
    </p>
    <p>
      Stars: 
      2
    </p>
    <button
      onClick={[Function]}
      type="button"
    >
      Add Bookmark
    </button>
  </div>
</div>
`;
